### ISSUE TITLE
[Question] How to properly cluster dense regions in 1D dataset?

### ISSUE THREAD

**Issue by edugmes** [scikit-learn-contrib/hdbscan#485]
My dataset below shows product sales per price [link to download dataset csv](https://drive.google.com/file/d/1OlKdkB7kocuTBWz-07d8SiQ8IvVt2_RX/view?usp=sharing):
```
     price      quantity
0    5098.0        20
1    5098.5        40
2    5099.0        10
3    5100.0        90
4    5100.5        20
..      ...       ...
290  5247.0       150
291  5247.5        30
292  5248.0       150
293  5248.5        20
294  5249.0        55

[295 rows x 2 columns]
```
**What I want to achive is clustering the dense regions (rectangles below) using HDBSCAN and sklearn**. We have four regions, but regions 3 and 4 could also be grouped into a big region, which would lead to only 3 regions on the entire dataset by changing the parameters **min_cluster_size** and **min_samples** in the function call.
![image](https://user-images.githubusercontent.com/6867503/128177271-dee4ba93-a3e5-44d9-95e0-51458ef60855.png)
And here is my code:
```
import hdbscan
import plotly.express as px
import pandas as pd
import numpy as np

data = pd.read_csv('data_set.csv')

price = group['price'].values.flatten()
price = price[:,np.newaxis]
weight = group['quantity'].values.flatten()

kde = KernelDensity(kernel='gaussian', bandwidth=1.5).fit(price,sample_weight=weight)
#the 85000 multiplication factor is only for visualization purposes of the density regions
data['prob'] = np.exp(kde.score_samples(price))*85000

fig = px.line(data,x='price',y='prob')
fig.add_bar(x=data['price'],y=data['quantity'])
fig.show()
```
![image](https://user-images.githubusercontent.com/6867503/128177306-d988efec-70d3-421a-9ae9-2b8fd8007451.png)
```
data = data[['price','quantity']]

clusterer = hdbscan.HDBSCAN(min_cluster_size=4,min_samples=8)
clusterer.fit(data)
data['cluster'] = clusterer.labels_

fig = px.bar(data,x='price',y='quantity',color='cluster',orientation='v')
fig.show()
```
![image](https://user-images.githubusercontent.com/6867503/128177315-299ef8e4-2eb4-4aea-92f2-6011ce29fb27.png)
**The problem is the result, the clustering did not work as I expected (picture above _versus_ picture below taken from [here](https://towardsdatascience.com/a-gentle-introduction-to-hdbscan-and-density-based-clustering-5fd79329c1e8))**. It clustered only the the amplitudes (quantity), but not the dense regions (amplitudes combined with price range). What am I missing in the code? 
![image](https://user-images.githubusercontent.com/6867503/128177333-83929768-16e3-460e-9695-87da6a48469a.png)

Things I've tried before calling HDBSCAN function:
- normalizing the data in both axis by subtracting each value with the dataset mean and divide by its standard deviatino
- swapping x-y axis 

Any help would be appreciated. I'm kind of lost in this code, as I thought by reading the documentation that it would be straight forward for this particular problem, as HDBSCAN deals great with density and noise.
